homework/index.js

```diff
@@ -30,15 +32,116 @@
     return elem;
   }
 
+  function makeHeader(repos) {
+    const root = document.getElementById('root');
+    const header = createAndAppend('header', root, { class: 'header' });
+    createAndAppend('h3', header, { text: 'HYF Repositories' });
+    const selectMenu = createAndAppend('select', header, { class: 'repo-selector' });
+    const optionsArr = [];
+    for (let i = 0; i < repos.length; i++) {
+      const option = {
+        name: repos[i].name,
+        index: i,
+      };
+      optionsArr.push(option);
+    }
+    optionsArr.map(element =>
+      createAndAppend('option', selectMenu, {
+        value: element.index,
+        text: element.name,
+        id: element.name,
+      }),
+    );
+    return header;
+  }
+
+  function makeTable(repo) {
+    const leftDiv = document.querySelector('.leftDiv');
+    const table = createAndAppend('table', leftDiv);
+    const tBody = createAndAppend('tbody', table);
+    const trRepo = createAndAppend('tr', tBody);
+    createAndAppend('td', trRepo, { text: 'repo:' });
+    const tdRepoLink = createAndAppend('td', trRepo);
+    createAndAppend('a', tdRepoLink, {
+      text: repo.name,
+      href: repo.html_url,
+      target: '_blank',
+    });
+    if (repo.description !== null) {
+      const trDescription = createAndAppend('tr', tBody, {});
+      createAndAppend('td', trDescription, { text: 'Description:' });
+      createAndAppend('td', trDescription, { text: repo.description });
+    }
+    const trForks = createAndAppend('tr', tBody);
+    createAndAppend('td', trForks, { text: 'Forks:' });
+    createAndAppend('td', trForks, { text: repo.forks_count });
+    const trUpdate = createAndAppend('tr', tBody);
+    createAndAppend('td', trUpdate, { text: 'Updated:' });
+    createAndAppend('td', trUpdate, {
+      text: new Date(repo.updated_at).toLocaleString(),
+    });
+  }
+
+  function makeContributorsList(repos) {
+    const rightDiv = document.querySelector('.rightDiv');
+    createAndAppend('p', rightDiv, { class: 'contributor-header', text: 'Contributions' });
+    const ul = createAndAppend('ul', rightDiv, { class: 'contributor-list' });
+    repos.forEach(contributor => {
+      const li = createAndAppend('li', ul, {
+        class: 'contributor-item',
+        tabindex: '0',
+        'aria-label': contributor.login,
+      });
+      createAndAppend('img', li, {
+        src: contributor.avatar_url,
+        class: 'contributor-avatar',
+        height: '48',
+        alt: 'profile photo of contributor',
+      });
+      const contributorData = createAndAppend('div', li, { class: 'contributor-data' });
+      createAndAppend('div', contributorData, { text: contributor.login });
+      createAndAppend('div', contributorData, {
+        class: 'contributor-badge',
+        text: contributor.contributions,
+      });
+      li.onclick = () => {
+        window.open(contributor.html_url, '_blank');
+      };
+    });
+  }
+
+  function makeHtmlOfContents(repos, index) {
```

Comment:
This function has no use for the complete array of repositories. It deals with just one repository (the selected one). In general, it is best to just give functions the exact data they need and nothing more (on a 'need to know' basis). This helps to make your code more robust as there is less chance of unintended data corruption. You could define this function like this:

```js
function  makeHtmlOfContents(selectedRepo) {
  ...
}
```

and call it like this:

```js
makeHtmlOfContents(repos[selectMenu.selectedIndex]);
```

Reply:
yes I applied and .. I don't know why I wrote that much code before..